Ledger guide (#572)
w3f/polkadot-wiki: docs/kusama-ledger.md

---
id: kusama-ledger
title: Using Ledger Devices
sidebar_label: Ledger Devices
---

Kusama has a [Ledger][] application that is compatible with the Ledger Nano S and Ledger Nano X

> devices. The Ledger devices are hardware wallets that keep your secret secured on a physical device

Reviewer: ```suggestion
devices. The Ledger devices are hardware wallets that keep your private key secured on a physical device
```

> that does not get exposed to your computer or the internet. It is strongly recommended to use a

Reviewer: ```suggestion
that does not get directly exposed to your computer or the internet. It is strongly recommended to use a
```

hardware wallet if you are managing a significant amount of funds.

The Kusama application allows you to manage Kusama's native asset, the KSM. It supports most of the
available transaction types of the network (a notable exception is the "Batch" transaction from the

> Utilty pallet).

Reviewer: ```suggestion
Utility pallet).
```

## Requirements

This is a list of what you will need before starting:

- A Ledger Nano S or a Ledger Nano X.

> - The latest firmware installed. (At the time of writing this was 1.6)

Reviewer 2: ```suggestion
- The latest firmware installed (at the time of writing this was 1.6).
```

- Ledger Live is installed.
- A web browser is installed that you can use to access [Polkadot-JS Apps][].

## Installing the Ledger Application

### Using Ledger Live

> > NOTE: If you are reading this notice the Kusama application has not been approved for Ledger Live

Reviewer: I assume we should get rid of this notification...

> yet. If you would like to proceed with the pre-release installation please follow the "Using the

> > Developer Release" instruction below. Otherwise hold tight, the application should be approved

Reviewer 2: ```suggestion
> Developer Release" instructions below. Otherwise hold tight, the application should be approved
```

> > soon!

Reviewer 2: ```suggestion
> soon.
```

- Open the "Manager" tab in Ledger Live.
- Connect and unlock your Ledger device.
- If asked, allow the manager on your device by pressing the right button.
- Find Kusama in the app catalog.
- Click the "Install" button of the app.
- An installation window appears.

> - Your device will display "Processing…"

Reviewer 2: ```suggestion
- Your device will display "Processing…".
```

Reviewer: Should there be a period if there is already an ellipsis inside quotes?  I feel like this is an "American vs British English" thing...

Reviewer 2: Huh - good question. I normally do this myself when writing (that's what I grew up learning).

executable in your shell by typing the command `chmod +x install_app.sh`.

Using `install_app.sh` help command will show you the available options:

```zsh
$ ./install_app.sh --help
Zondax Installer [Kusama-1.2011.1] [Warning: use only for test/demo apps]
  load    - Load Kusama app
  delete  - Delete Kusama app
  version - Show Kusama app version
```

Next, you must make sure your Ledger device is plugged in and unlocked and you're using the latest
firmware (1.6 at the time of writing). If everything is prepared, then type `install_app.sh load`
and accept the prompts on your Ledger device to install the application.

First it will prompt you to allow an unsafe manager. Navigate all the way to the left and click both
buttons to allow the unsafe manager to install the Kusama app.

After some processing time, the screen of your device will update to say "Install app Kusama".
Again, navigate all the way to the right, verify the Identifier hash matches the one that is printed
in your terminal. Click both buttons on "Perform Installation" to install the application. It will
ask again for your Pin code and you should enter it in the device.

> At the end of the process you should the newly installed Kusama application on the device. Great

Reviewer 2: ```suggestion
At the end of the process you should the newly installed Kusama application on the device.
```

> At the end of the process you should the newly installed Kusama application on the device. Great

Reviewer: ```suggestion
At the end of the process you should have the newly installed Kusama application on the device. Great
```

> job!

Reviewer 2: ```suggestion
```

## Using on Polkadot-JS Apps

### Loading Your Account

> [Polkadot-JS Apps][] already has an integration with the Ledger application so that your device will

Reviewer: We'll have to talk to Jaco about turning this off at some point and edit the wiki.

work with the browser interface after installation. The functionality is currently gated behind a
feature setting that you will need to turn on.

In order to turn on the interoperability with the Kusama Ledger application, go to the "Settings"
tab in Polkadot-JS Apps. Find the option for attachting Ledger devices and switch the option from
the default "Do not attach Ledger devices" to "Attach Ledger via WebUSB".

![](assets/ledger.png)

Click "Save" to keep your settings.

Now when you go to the "Accounts" tab you will see a new button that says "Query Ledger". Ensure
that your Ledger device is unlocked and you have navigated into the Kusama application, then click

> this button.

Reviewer 2: ```suggestion
the "Query Ledger" button.
```
I think we should be descriptive about buttons as some people might be visually impaired or have trouble with loading the image onto the screen.

![](assets/ledger-2.png)

> You will not get any feedback from the web page, but if all went fine you should be able to scroll

Reviewer: ```suggestion
You will not get any feedback from the web page, but if all went well you should be able to scroll
```

down and find a new account on the page named "Ledger". Under the type column, the account will also
show "ledger".

![](assets/ledger-3.png)

You can now use this account to interact with Kusama on Polkadot-JS Apps and it will prompt your
ledger for confirmation when you initiate a transaction.

### Confirming the Address on your Device

On the "Accounts" tab, find your Ledger-connected account (the one called "Ledger"). Click on the
three vertical dots at the left of the row. This will open a new menu, here you can click the "Show
address on hardware device" option to display the address on your device.

![](assets/ledger-4.png)

Here you can scroll through and make sure the address matches to what is displayed on Polkadot-JS
Apps.

### Checking Balance of Your Account

> There are a couple methods to check the balance of your account. You can use Polkadot-JS Apps or you

Reviewer: Maybe add links here?

> There are a couple methods to check the balance of your account. You can use Polkadot-JS Apps or you

Reviewer: ```suggestion
There are a few methods to check the balance of your account. You can use Polkadot-JS Apps or you
```

can use a block explorer like Polkascan.

#### Using Polkadot-JS Apps

Once you have your account loaded on the "Accounts" tab it should show a row with your Ledger
account. At the far right of the row is located your account's KSM balances. If you click the menu
open it will show details of your balance such as locks or reserve amounts.

### Using a Block Explorer

You can also use a block explorer such as [Polkascan][] or [Subscan][] to check the KSM balance of
your account. On these block explorers you can search for you account in the search box and click on

> it to reveal the details. It will show your KSM balance as well as details about your bonded or

Reviewer 2: ```suggestion
it to reveal the details. It will show your KSM balance as well as information about your bonded or
```

locked amounts.

### Sending a Transfer

If you would like to send a transfer from your account housed on the Ledger device, the easiest

> method is to use Polkadot-JS Apps.

Reviewer 2: ```suggestion
method is to use [Polkadot-JS Apps][].
```

> - Click on the "Transfer" button on the left navigation menu.

Reviewer 2: There isn't a left navigation menu anymore... =)

- In the top input, select "Ledger" as your sending account.
- In the second input, select the account that you want to transfer funds to.
- In the third input, enter the amount of KSM you want to transfer.
- Click the "Make Transfer" button.
- Confirm the transaction on your device.
- A green success notification will be displayed when the transaction is included in a block.

### Receiving a Transfer

In order to receive a transfer on the account that is stored on your Ledger device, you will need to
provide the sender (i.e. the payer) your address. You can copy your address to the clipboard after
loading your device on Polkadot-JS Apps by clicking the account icon or by highlighting the address

> and pressing ctrl-c. Then you will provide your address to the payer so they can send a transfer.

Reviewer 2: ```suggestion
and pressing ctrl-c. Then you will provide your address to the payer so they can send a transfer.
```
Ah - but on some computers, ctrl-c might be command c.
